File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_val = 0
    card_id = 0
    answer = 0
    for idx, item in enumerate(sizes):
        w, h = item[:]
        if w > max_val:
            max_val = w
            card_id = idx
        if h > max_val:
            max_val = h
            card_id = idx
    
    _max_val = min(sizes[card_id])
    for idx, item in enumerate(sizes):
        if idx == card_id:
            continue
        w, h = item[:]
        
        min_val = min(w, h)
        
        if _max_val < min_val:
            _max_val = min_val
    
    answer = max_val * _max_val
        
    
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        
        int max_val = 0;
        int pair_val = 0;
        
        for (int[] size : sizes){
            int a = size[0];
            int b = size[1];
            int temp = 0;
            if (a<b){
                temp = a;
                a = b;
                b = temp;
            }
            if (max_val < a){
                max_val = a;
                pair_val = b;
            }
        }
            
        for (int[] size : sizes){
            int a = size[0];
            int b = size[1];
            int temp = 0;
            if (a<b){
                temp = a;
                a = b;
                b = temp;
            }
            
            if (pair_val < b){
                pair_val = b;
            }   
        }
        
        
        int answer = 0;
        answer = pair_val * max_val;
        return answer;
    }
}